src/pymodaq_plugins_newport/daq_move_plugins/daq_move_Newport_XPS_Q8.py

```diff
@@ -0,0 +1,294 @@
+        else:
+            if (errorCode == -2):
+                self._plugin.emit_status(ThreadCommand('Update_Status', [f'{APIName} : TCP timeout']))
+            if (errorCode == -108):
+                self._plugin.emit_status(ThreadCommand('Update_Status', [f'{APIName} : The TCP/IP connection was closed by an administrator']))
+        self.closeTCPIP()
+
+    def closeTCPIP(self):
+        """Call the method to close the socket."""
+        self.myxps.TCP_CloseSocket(self.socketId)
+        
+    def getPosition(self):
+        """Returns current the position"""
+        [errorCode, currentPosition] = self.myxps.GroupPositionCurrentGet(self.socketId, self._full_positionner_name, 1)
+        if (errorCode != 0):
+            self.displayErrorAndClose(errorCode, 'GroupPositionCurrentGet')
+            sys.exit()  
+        else:
+            return(float(currentPosition))
+    
+    def moveAbsolute(self, value):
+        """Moves the stage to the position value."""
+        if (self.socketId != -1):
+            [errorCode, returnString] = self.myxps.GroupMoveAbsolute(self.socketId, self._full_positionner_name, [value])
+            if (errorCode != 0):
+                self.displayErrorAndClose(errorCode, 'GroupMoveAbsolute')
+            
+    
+    def moveRelative(self, value):
+        """Moves the stage to value relative to it's current position."""
+        if (self.socketId != -1):
+            [errorCode, returnString] = self.myxps.GroupMoveRelative(self.socketId, self._full_positionner_name, [value])
+            if (errorCode != 0):
+                self.displayErrorAndClose(errorCode, 'GroupMoveRelative')
+                
+    def moveHome(self):
+        """Moves the stage to it's home"""
+        if (self.socketId != -1):
+            [errorCode, returnString] = self.myxps.GroupHomeSearch(self.socketId, self._group)
+            if (errorCode != 0):
+                self.displayErrorAndClose(errorCode, 'GroupHomeSearch')
+
+    
+    def setGroup(self, group:str):
+        self._group = group
+        self._full_positionner_name = f'{group}.{self._positionner}'
+    
+    def setPositionner(self, positionner:str):
+        self._positionner = positionner
+        self._full_positionner_name = f'{self._group}.{positionner}'
+    
+    def setIP(self, IP:str):
+        """Sets a new IP address. Does not automatically try to connect with it, call retryConnection."""
+        self._ip = IP
+    
+    def setPort(self, port:int):
+        """Sets a new port. Does not automatically try to connect with it, call retryConnection."""
+        self._port = port
+        
+    def retryConnection(self):
+        """Closes the connection and runs the init sequence again.
+        Called by the plugin after any change to the IP address or port parameters."""
+        self.closeTCPIP()
+        self._initCommands()
+    
+class DAQ_Move_Newport_XPS_Q8(DAQ_Move_base):
```

Comment:
For the class name I would rather use:
 DAQ_Move_NewportXpsQ8

or directly
DAQ_Move_XpsQ8

Reply:
Changed to DAQ_Move_XpsQ8 on commit [8cd0fe3](https://github.com/PyMoDAQ/pymodaq_plugins_newport/pull/6/commits/8cd0fe3b456c37e5a4d1f27d5da8fdf59554b56d)